easypy/concurrency.py

```diff
@@ -599,6 +599,10 @@ def chain(self):
         "Chain the iterables contained by this MultiObject"
         return self.__class__(chain(*self))
 
+    def zip_with(self, *collections):
+        assert all(len(coll) == len(self) for coll in collections), "All collection must be of length %s" % len(self)
```

Comment:
I think that's dangerous... at the very least `self.items` should all be paired, to avoid subtle mistakes